odoo/tests/common.py

```diff
@@ -2003,14 +2094,33 @@ def assertURLEqual(self, test_url, truth_url, message=None):
             message,
         )
 
-    def url_open(self, url, data=None, files=None, timeout=12, headers=None, allow_redirects=True, head=False):
+    def url_open(self, url, data=None, files=None, timeout=12, headers=None, json=None, params=None, allow_redirects=True, method: str | None = None, head=False, options=False, post=False):
```

Comment:
In a separate PR remove `head`, `options` and `post` parameters in favour of method (method could still be inferred through other parameters such as `data`, `files` or `json`